cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py

```diff
@@ -0,0 +1,387 @@
+import sys
+from PyQt5.QtWidgets import *
+from PyQt5.QtCore import *
+from PyQt5.QtGui import *
+from cc3d.player5 import Configuration
+from cc3d.player5.Plugins.ViewManagerPlugins import ui_demo_browser
+import weakref
+import re
+import time
+from pathlib import Path
+from collections import Counter
+from cc3d.player5.styles.SyntaxHighligher import *
+
+MISSING_FILE_MESSAGE = "This demo doesn't have that kind of file available to preview."
+DESCRIPTION_FNAME = 'README.txt'
+INDEX_FILE_NAME = '.demoindex'
+DELIMITER = ','
+
+
+def getDemoRootPath() -> Path | None:
+    """
+    Credit to https://stackoverflow.com/a/74532549/16519580 user Chris.
+    """
+
+    if Configuration.check_if_setting_exists("DemosPath"):
+        demosPath = Configuration.getSetting("DemosPath")
+        demosPath = Path(demosPath)
+        #Ensure path is not the root dir
+        if len(list(demosPath.parents)) > 1 and demosPath.exists():
+            return demosPath
+
+    #Start at location of python.exe and check in every
+    #parent directory for Demos until we hit the root.
+    startDir = Path(__file__)
+
+    for currPath in [startDir] + list(startDir.parents):
+        demosPath = currPath.joinpath('CompuCell3D/Demos')
+        if demosPath.exists():
+            Configuration.setSetting("DemosPath", str(demosPath))
+            return demosPath
+
+    return None #No demos here!
+
+def getDemoList() -> list[Path]:
+    rootPath = getDemoRootPath()
+    globResults = rootPath.rglob('*.cc3d')
```

Comment:
I worry that `rglob` has a small chance of searching the user's entire file system if their directories are arranged a certain way. Is there a good way to limit the maximum search depth or cause a timeout?